ui/tests/integration/components/search-select-test.js

```diff
@@ -0,0 +1,164 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { create } from 'ember-cli-page-object';
+import { typeInSearch, clickTrigger } from 'ember-power-select/test-support/helpers';
+import Service from '@ember/service';
+import { render } from '@ember/test-helpers';
+import { run } from '@ember/runloop';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+import searchSelect from '../../pages/components/search-select';
+
+const component = create(searchSelect);
+
+const storeService = Service.extend({
+  query(modelType) {
+    return new Promise((resolve, reject) => {
+      switch (modelType) {
+        case 'policy/acl':
+          resolve([{ id: '1', name: '1' }, { id: '2', name: '2' }, { id: '3', name: '3' }]);
+          break;
+        case 'policy/rgp':
+          reject({ httpStatus: 403 });
+          break;
+        case 'identity/entity':
+          resolve([{ id: '7', name: 'seven' }, { id: '8', name: 'eight' }, { id: '9', name: 'nine' }]);
+          break;
+        default:
+          reject({ httpStatus: 404 });
+          break;
+      }
+      reject({ httpStatus: 404 });
+    });
+  },
+});
+
+module('Integration | Component | search select', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function() {
+    run(() => {
+      this.owner.unregister('service:store');
+      this.owner.register('service:store', storeService);
+    });
+    component.setContext(this);
```

Comment:
Something I learned when I was removing jQuery - the setContext / removeContext isn't needed with the new test style so we can remove this and the afterEach hook below. (the no-jquery branch removes them from all of the other tests).